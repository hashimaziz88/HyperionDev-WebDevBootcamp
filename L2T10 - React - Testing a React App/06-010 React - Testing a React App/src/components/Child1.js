import React from 'react'

export default function Child1(props) {
    
  return (
    <div className='container'>
        <p>{props.name}</p>
        <button onClick={()=> props.setName("Jason")} >change</button>
        </div>
  )
}
