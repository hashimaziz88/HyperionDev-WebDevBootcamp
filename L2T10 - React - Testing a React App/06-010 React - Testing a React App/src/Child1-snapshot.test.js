import React from 'react';
import renderer from 'react-test-renderer'; // Install react-test-renderer
import Child1 from './components/Child1'; // Import your component

test('Child1 component snapshot', () => {
  const component = renderer.create(<Child1 />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
