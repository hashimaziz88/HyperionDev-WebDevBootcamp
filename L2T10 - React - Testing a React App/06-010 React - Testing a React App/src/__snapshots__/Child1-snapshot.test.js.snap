// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Child1 component snapshot 1`] = `
<div
  className="container"
>
  <p />
  <button
    onClick={[Function]}
  >
    change
  </button>
</div>
`;
